fedimint-bitcoind/src/lib.rs

```diff
@@ -198,22 +198,28 @@ impl DynBitcoindRpc {
     pub fn spawn_block_count_update_task(
```

Comment:
IMO BitcoindRpc should have method that returns `Receiver<Option<u64>>`

and spawn_block_count_update_task sends to that channel using `send_if_modified`.

last_block_count kind of merges into current value of watch channel

Reply:
@maan2003 Returning receiver forces the caller to use a specific type of a receiver and can't handle e.g. broadcasting to multiple channels. It also gives us no hook to put logging when the value is updated. Also with `on_update` the caller can handle the initial value (`Option`, or `0`, or whatever).

It's possible, but would require another spawned task. 

So `on_update` is a bit more flexible, at the cost of a slightly worse ergonomics.

It's true that `last_block_count` works kind of like a `send_if_modified`.